sql/parser/datum.go

```diff
@@ -31,26 +31,26 @@ import (
 
 var (
 	// DummyBool is a placeholder DBool value.
-	DummyBool = DBool(false)
+	DummyBool = Datum(DBool(false))
```

Comment:
Changed. And I got rid of the non-redundant assertions below that the various `Datum` implementations can be assigned to `Datum`.